README.md

```diff
@@ -89,11 +89,15 @@ Ginkgo adds the following additional switches to control what is being built:
     Ginkgo's documentation. The default is `OFF`.
 *   `-DGINKGO_EXPORT_BUILD_DIR={ON, OFF}` adds the Ginkgo build directory to the
     CMake package registry. The default is `OFF`.
-*   `-DCMAKE_INSTALL_PREFIX=path` sets the installation path for `make install`.
-    The default value is usually something like `/usr/local`
 *   `-DGINKGO_VERBOSE_LEVEL=integer` sets the verbosity of Ginkgo.
     * `0` disables all output in the main libraries,
     * `1` enables a few important messages related to unexpected behavior (default).
+*   `-DCMAKE_INSTALL_PREFIX=path` sets the installation path for `make install`.
+    The default value is usually something like `/usr/local`
+*   `-DCMAKE_BUILD_TYPE=type` specifies which configuration will be used for
+    this build of Ginkgo. There are no default. Supported values are CMake's
+    standard build types such as `DEBUG` and `RELEASE` and the Ginkgo specific 
```

Comment:
What is the default value of this variable ?

Reply:
As stated, `there are no default`.

Reply:
Maybe `RelWithDebInfo` as default is ideal ?

Reply:
There is a default, and default is "" (empty string), that is also an option, in which case none of the variables `CMAKE_SOMETHING_<CONFIG>` get merged into their `CMAKE_SOMETHING` variants.

Here are the default initializations for the GNU compiler (which are also inherited by the Clang compiler) for various configurations: https://github.com/Kitware/CMake/blob/master/Modules/Compiler/GNU.cmake#L50

Looking at these, `RelWithDebInfo` should not be interpreted as "a good default which provides both optimization and debug capabilities", but rather as "an option to debug problems that arise from enabling optimizations", which is probably not something very useful for the average user. I would go for `Release` as the default, since that is what will be the first thing users want to try - how fast does this thing work, and instruct them to build and link with the debug version if they think they encountered a bug.

Reply:
Thanks for looking, I did know of empty (`""`) but was never sure what it meant. So basically by default there are no particular flags specified, is that correct?

I agree with making Release default. I will do this as part of this PR.